### PR TITLE
Eos timer

### PR DESCRIPTION
**Pull request by frostbyte73**

### CODE REVIEW

File: pkg/pipeline/watch.go [hunk 1]
```diff
@@ -121,6 +121,7 @@ func (c *Controller) messageWatch(msg *gst.Message) bool {
 	switch msg.Type() {
 	case gst.MessageEOS:
 		logger.Infow("EOS received")
+		c.eosTimer.Stop()
```
Comment (davidzhao) [inline pkg/pipeline/watch.go:124]: nil check? possible EOS is sent by something else?

Comment (frostbyte73) [reply pkg/pipeline/watch.go:124]: shouldn't be possible but might as well add it

File: pkg/pipeline/controller.go [hunk 2]
```diff
@@ -405,7 +407,12 @@ func (c *Controller) SendEOS(ctx context.Context) {
 
 		case livekit.EgressStatus_EGRESS_ENDING,
 			livekit.EgressStatus_EGRESS_LIMIT_REACHED:
-			go c.p.SendEOS()
+			go func() {
+				c.eosTimer = time.AfterFunc(time.Second*30, func() {
+					c.OnError(errors.ErrPipelineFrozen)
+				})
+				c.p.SendEOS()
```
Comment (biglittlebigben) [inline pkg/pipeline/controller.go:414]: Will this kill in process uploads? We have many examples of cases where upload takes more than 30 seconds (a 3h file could be several GB, and even for shorter sessions, we have cases where S3 is just slow). We should not charge customers for that time, but we should keep the upload going.

Comment (frostbyte73) [reply pkg/pipeline/controller.go:414]: the media won't be available if the eos is never received

Comment (frostbyte73) [reply pkg/pipeline/controller.go:414]: (unless it's HLS, in which case the playlist will not be closed)

**Review by davidzhao — APPROVED** [review]

File: pkg/pipeline/controller.go (hunk 2 above)
Comment (biglittlebigben) [reply pkg/pipeline/controller.go:414]: My question is about calling `c.OnError(errors.ErrPipelineFrozen)` after 30s.

**Pull request merged**